src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
```

Comment:
This is fine as-is, but you could shorten it to something like
```
const data = await (fetch(url, {
  method: "GET",
  headers: { accept: "application/json" }
})).json();
```

I didn't test that, but that or something VERY similar would save you a few lines.

Reply:
I'm not sure where to put the repsonse here and can't get it to work.

I tried
``` js
async fetchJokes() {
    this.checkInput(this.state.searchTerm)
    try {
      const myData = await (fetch(url, {
        method: "GET", headers: headers
      })).json();
      this.setState({
        jokeData: { results: myData.results },
        loadingState: LOADING_STATE.loaded
      });
```
So I'll park it for now and come back to it.